### PR TITLE
Tox on Windows

### PR DESCRIPTION
**Pull request by gpotter2**
Try use tox for Appveyor builds.

Note:
- I had to move `windump.exe` to `C:/Program Files/Windump/` because due to some windows restrictions (you can't add & access files from `C:/Windows/system32` when running on x64, but it should be `C:\Windows\SysWOW64`)
- I had to unify all "test_travis...." files into the `.uts` file, so that we can launch it from everywhere (don't bother about the path import)
- I had some issues testing it, as I had often a single networking test failing. I added to all of them an auto-retry system as it already existed in the code (+ added some missing `timeout=...`). (I had some tests hanging and had to retry lots of time... I may move them to a different PR)

### CODE REVIEW

**Comment by gpotter2** [comment]
Now we are getting somewhere, this was way harder than I expected...

Remarks:
- The "Stabilize all networking tests" + "retry_test" x 2 tests are not needed in this PR, and may be moved. However, it would require extra work that I'm lazy to give in, so I would appreciate if it could be merged together...
- 2 tests were marked as "require_gui" because they prompt a [UAC](https://en.wikipedia.org/wiki/User_Account_Control), that is bypassed normally, but not with tox (for some reason).
- I had to tweak the way `route_add_loopback` works so that it works in all cases, without depending on previous tests + how many interfaces the machine has

File: test/tls/tests_tls_netaccess.uts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 % TLS session establishment tests
 
+~ crypto needs_root
```
Comment (gpotter2) [inline test/tls/tests_tls_netaccess.uts]: Added this on top of the file to make it global

**Comment by gpotter2** [comment]
@guedou This is finally ready to be reviewed!

File: scapy/tools/UTscapy.py [hunk 2]
```diff
@@ -843,6 +843,9 @@ def main(argv):
             from scapy.arch.windows import route_add_loopback
             route_add_loopback()
 
+        # Add SCAPY_ROOT_DIR environment variable, used for tests
+        os.environ['SCAPY_ROOT_DIR'] = os.environ.get("PWD", os.getcwd())
```
Comment (guedou) [inline scapy/tools/UTscapy.py]: Shall we remove this variable at the end of UTScapy execution?

File: test/edns0.uts [hunk 3]
```diff
@@ -60,10 +60,32 @@ raw(tlv) == b'\x00\x05\x00\x04\x00\x11"3'
 tlv = EDNS0TLV(optcode=2, optdata="")
 raw(tlv) == b'\x00\x02\x00\x00'
 
+= Util - setup retry_test()
+
+def retry_test(func):
```
Comment (guedou) [inline test/edns0.uts]: It could be nice to make `retry_test()` directly available from UTScapy.

File: scapy/tools/UTscapy.py (hunk 2 above)
Comment (gpotter2) [reply scapy/tools/UTscapy.py]: Done

Comment (p-l-) [reply scapy/tools/UTscapy.py]: Why would we do that? The environment variable won't be passed to other programs anyway when UTScapy terminates. Or did I miss something?

Comment (gpotter2) [reply scapy/tools/UTscapy.py]: It’s cleaner, as people may use UTscapy outside travis

Comment (p-l-) [reply scapy/tools/UTscapy.py]: I do not agree. You do not remove an environment variable set by the user.

**Comment by gpotter2** [comment]
The variable we are talking about is set by UTscapy and only used during tests. It was previously set in Travis.yml or in the .bat but because it was static in them, we now set it dynamically

File: test/edns0.uts (hunk 3 above)
Comment (gpotter2) [reply test/edns0.uts]: Done

**Comment by p-l-** [comment]
@gpotter2 yep, I removed my comment because I saw that just after posting my answer.

**Pull request merged**